code/modules/projectiles/projectile.dm

```diff
@@ -170,15 +170,29 @@
 			permutated.Add(A)
 		return FALSE
 	else
-		if(A && A.density && !ismob(A) && !(A.flags_1 & ON_BORDER_1)) //if we hit a dense non-border obj or dense turf then we also hit one of the mobs on that tile.
+		if(A && A.density && !ismob(A) && !(A.flags_1 & ON_BORDER_1) && !istype(A,/obj/machinery) && !istype(A,/obj/structure/)) //if we hit a dense non-border obj or dense turf then we also hit one of the mobs on that tile.

```

Comment:
Why is /obj/structure in here when you never try to hit them? Also this ```istype(A, /obj/machinery)``` should be moved before the ```!A.flags_1 & ON_BORDER_1)``` check, for flow's sake.

Reply:
Attacks on tiles that had grilles or windows would also result in an machinery on that tile being hit. So the projectile would effectively hit 2 things.